engine/src/test/java/com/redhat/cloud/notifications/processors/email/EmailProcessorTest.java

```diff
@@ -0,0 +1,400 @@
+        // will simulate that is returned from the template repository.
+        final String subjectData = "test-missing-subscribers-subject-data";
+        final String bodyData = "test-missing-subscribers-body-data";
+
+        final Template subjectTemplate = new Template();
+        subjectTemplate.setData(subjectData);
+
+        final Template bodyTemplate = new Template();
+        bodyTemplate.setData(bodyData);
+
+        final InstantEmailTemplate instantEmailTemplate = new InstantEmailTemplate();
+        instantEmailTemplate.setSubjectTemplate(subjectTemplate);
+        instantEmailTemplate.setBodyTemplate(bodyTemplate);
+
+        Mockito.when(this.templateRepository.findInstantEmailTemplate(event.getEventType().getId())).thenReturn(Optional.of(instantEmailTemplate));
+
+        // Mock the template instances that will be returned from the compiling
+        // operation.
+        final TemplateInstance subjectTemplateInstance = Mockito.mock(TemplateInstance.class);
+        final TemplateInstance bodyTemplateInstance = Mockito.mock(TemplateInstance.class);
+        Mockito.when(this.templateService.compileTemplate(subjectData, "subject")).thenReturn(subjectTemplateInstance);
+        Mockito.when(this.templateService.compileTemplate(bodyData, "body")).thenReturn(bodyTemplateInstance);
+
+        // Mock the rendered contents that will be returned from the rendering
+        // operation.
+        final String stubbedRenderedSubject = "test-missing-subscribers-rendered-subject";
+        final String stubbedRenderedBody = "test-missing-subscribers-rendered-body";
+        Mockito.when(this.templateService.renderTemplate(event.getEventWrapper().getEvent(), subjectTemplateInstance)).thenReturn(stubbedRenderedSubject);
+        Mockito.when(this.templateService.renderTemplate(event.getEventWrapper().getEvent(), bodyTemplateInstance)).thenReturn(stubbedRenderedBody);
+
+        // Do not return any subscribers for this test.
+        Mockito.when(this.emailSubscriptionRepository.getSubscribersByEventType(event.getOrgId(), event.getEventType().getId(), EmailSubscriptionType.INSTANT)).thenReturn(List.of());
+
+        // Call the processor under test.
+        this.emailProcessor.process(event, endpoints);
+
+        // Verify that the compilation functions were called.
+        Mockito.verify(this.templateService, Mockito.times(1)).compileTemplate(subjectData, "subject");
+        Mockito.verify(this.templateService, Mockito.times(1)).compileTemplate(bodyData, "body");
+
+        // Verify that the rendering functions were called.
+        Mockito.verify(this.templateService, Mockito.times(1)).renderTemplate(event.getEventWrapper().getEvent(), subjectTemplateInstance);
+        Mockito.verify(this.templateService, Mockito.times(1)).renderTemplate(event.getEventWrapper().getEvent(), bodyTemplateInstance);
+
+        // Verify that, since there were no subscribers to notify, the event
+        // was ignored.
```

Comment:
I probably approved something wrong in `EmailProcessor`. If the users preferences are ignored, we should send the notification to the email connector when there are no subscribers.

Reply:
I will open a PR then to fix that: if we don't find any subscribers in the database, we still keep going with the execution, then:

https://github.com/RedHatInsights/notifications-backend/blob/e65f278eeb83d0aada5c00907c70405ee38f26af/engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailProcessor.java#L67-L73

Reply:
As discussed in Slack, we'll deal with that in a subsequent PR.